tools/mo/openvino/tools/mo/utils/version.py

```diff
@@ -75,3 +73,15 @@ def get_simplified_ie_version(env=dict(), version=None):
     if m and len(m.groups()) == 3:
         return simplify_version(m.group(3))
     return simplify_version(version)
+
+
+def extract_version_hash(full_version: str):
+    """
+    extracts 11 character hash from the full version string
+    :param full_version:
+    :return:
+    """
+    res = re.findall(r'[-_]{1}([a-z0-9]{7,})', full_version)
+    if res is not None:
+        res = res[0][:7]
```

Comment:
In the meeting we seem to agree on the different approach that you extract the full hash for both IE and MO versions, find the minimum size between them, cut both version ups to this minimum and compare them

Reply:
add comparison by minimal number of hex digits